contrib/babelfishpg_tsql/src/dbcmds.c

```diff
@@ -301,9 +301,15 @@ getAvailDbid(void)
 	int16		dbid;
 	int16		start = 0;
 
+	if(GetUserId() != get_role_oid("sysadmin", true))
+		return InvalidDbid;
+
 	do
 	{
-		dbid = DirectFunctionCall1(nextval, CStringGetTextDatum("sys.babelfish_db_seq"));
+		RangeVar	*sequence = makeRangeVarFromNameList(stringToQualifiedNameList("sys.babelfish_db_seq"));
+		Oid			seqid = RangeVarGetRelid(sequence, NoLock, false);
+
+		dbid = nextval_internal(seqid, false);
```

Comment:
just to be safe, add assert before calling nextval_internal - `Assert(OidIsValid(seqid))`